pkg/apis/batch/validation/validation.go

```diff
@@ -177,29 +186,38 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 			containerNames.Insert(containerSpec.Name)
 		}
 		for i, rule := range spec.PodFailurePolicy.Rules {
+			rulePath := fldPath.Child("podFailurePolicy").Index(i)
+
 			if rule.Action != batch.PodFailurePolicyActionIgnore && rule.Action != batch.PodFailurePolicyActionTerminate {
-				path := fldPath.Child("template").Index(i).Child("onExitCodes").Child("action")
-				allErrs = append(allErrs, field.NotSupported(path, rule.Action, []string{string(batch.PodFailurePolicyActionIgnore), string(batch.PodFailurePolicyActionTerminate)}))
+				actionPath := rulePath.Child("action")
+				allErrs = append(allErrs, field.NotSupported(actionPath, rule.Action, []string{string(batch.PodFailurePolicyActionIgnore), string(batch.PodFailurePolicyActionTerminate)}))
 			}
 			if rule.OnExitCodes != nil {
+				onExitCodesPath := rulePath.Child("onExitCodes")
 				if rule.OnExitCodes.Operator != batch.PodFailurePolicyOnExitCodesOpIn && rule.OnExitCodes.Operator != batch.PodFailurePolicyOnExitCodesOpNotIn {
-					path := fldPath.Child("template").Index(i).Child("onExitCodes").Child("operator")
+					path := onExitCodesPath.Child("operator")
 					allErrs = append(allErrs, field.NotSupported(path, rule.OnExitCodes.Operator, []string{string(batch.PodFailurePolicyOnExitCodesOpIn), string(batch.PodFailurePolicyOnExitCodesOpNotIn)}))
 				}
 				if rule.OnExitCodes.ContainerName != nil && !containerNames.Has(*rule.OnExitCodes.ContainerName) {
-					path := fldPath.Child("template").Index(i).Child("onExitCodes").Child("containerName")
-					allErrs = append(allErrs, field.Invalid(path, *rule.OnExitCodes.ContainerName, "must be one of the specified container names"))
+					path := onExitCodesPath.Child("containerName")
+					allErrs = append(allErrs, field.Invalid(path, *rule.OnExitCodes.ContainerName, "must be one of the container names in the pod template"))
 				}
 			}
 			if rule.OnPodConditions != nil {
+				onPodConditionsPath := rulePath.Child("onPodConditions")
 				if rule.OnPodConditions.Operator != batch.PodFailurePolicyOnPodConditionsOpIn {
-					path := fldPath.Child("template").Index(i).Child("onPodConditions").Child("operator")
+					path := onPodConditionsPath.Child("operator")
 					allErrs = append(allErrs, field.NotSupported(path, rule.OnPodConditions.Operator, []string{string(batch.PodFailurePolicyOnPodConditionsOpIn)}))
 				}
+				for j, podConditionValue := range rule.OnPodConditions.Values {
+					valuePath := onPodConditionsPath.Child("Values").Index(j)
+					if podConditionValue != targetedForDisruption {
```

Comment:
Users are free to add any condition they want to pods, so we should allow them to use them from the job controller.

My previous comment was about having the same validation as the conditions in the pod status

https://github.com/kubernetes/kubernetes/blob/9525206bcb1b51e3cb11e5efc19f274fc9cfc6fd/pkg/apis/core/validation/validation.go#L4411

We probably need to add our new constant to the list in beta

Reply:
I see, I relaxed the validation to allow custom condition types, but check that they pass the checks in `apivalidation.ValidateQualifiedName`. 

This is essentially what the reference code does. We could add our condition to the referenced validation code for Beta, but I see an issue with this code in general: 
- the list of system pod conditions does not make any difference (maybe performance?) for validation, cause these names would pass the `ValidateQualifiedName` checks (verified).
- the list of system pod conditions is not maintained already, it is already missing `PodReasonUnschedulable` and `ContainersReady`.

Maybe we just have a cleanup PR to eliminate the loop over the system conditions?